### PR TITLE
Remove the `cmd` parameter from `Tool`, and `cliEnvSettingsRequiredTools` from `CliEnvSettings`

### LINKED ISSUE #67: Rethink 'CliEnvSettings' and 'defaultCliEnvSettings'
Currently, the `CliEnvSettings` data type is defined in the following way:

https://github.com/chshersh/iris/blob/0b9c50da0a277b298c62064929eb2f875c902841/src/Iris/Env.hs#L53-L71

And `defaultCliEnvSettings`:

https://github.com/chshersh/iris/blob/0b9c50da0a277b298c62064929eb2f875c902841/src/Iris/Env.hs#L78-L86

The idea of using `()` in the type signature was to rely on type-changing record updates. Unfortunately, the following code doesn't compile:

```haskell
cmdP :: Parser Cmd
cmdP = ...

appSettings :: Iris.CliEnvSettings Cmd ()
appSettings = Iris.defaultCliEnvSettings
    { Iris.cliEnvSettingsCmdParser = cmdP
    }
```

It fails with the error:

```
... location ...
    • Couldn't match type ‘()’ with ‘Cmd’
      Expected: CliEnvSettings Cmd ()
        Actual: CliEnvSettings () ()
    • In the expression:
        Iris.defaultCliEnvSettings {cliEnvSettingsCmdParser = cmdP}
      In an equation for ‘appSettings’:
          appSettings
            = Iris.defaultCliEnvSettings {cliEnvSettingsCmdParser = cmdP}
   |
26 | appSettings = Iris.defaultCliEnvSettings
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^...

... location ...
    • Couldn't match type ‘Cmd’ with ‘()’
      Expected: Options.Applicative.Types.Parser ()
        Actual: Options.Applicative.Types.Parser Cmd
    • In the ‘cliEnvSettingsCmdParser’ field of a record
      In the expression:
        Iris.defaultCliEnvSettings {cliEnvSettingsCmdParser = cmdP}
      In an equation for ‘appSettings’:
          appSettings
            = Iris.defaultCliEnvSettings {cliEnvSettingsCmdParser = cmdP}
   |
27 |     { cliEnvSettingsCmdParser = cmdP
   |                                 ^^^^
```

The problem here because the `cliEnvSettingsRequiredTools` is also parametrised by `cmd`. So to change the type, you need to update both fields like this:

```haskell
appSettings :: Iris.CliEnvSettings Cmd ()
appSettings = Iris.defaultCliEnvSettings
    { Iris.cliEnvSettingsCmdParser = cmdP
    , Iris.cliEnvSettingsRequiredTools = []
    }
```

Which is a shame. You need to set an extra field you don't care about all the time because you want to change the type of the CLI command.

It would be great to improve this interface. However, I don't have ideas at the moment. This requires some thinking 🤔

### PR DESCRIPTION
**Pull request by german1608**
Resolves #67 

* Removes the `cmd` parameter from `Tool` (and consequently, `ToolSelector`)
* Add the `need` function, which does pretty much this: https://github.com/chshersh/iris/blob/main/src/Iris/Env.hs#L123-L126

### Additional tasks

- [ ] Documentation for changes provided/changed
  > (no docs added since I'm unsure of the final signature of `need`)
- [x] Tests added
- [ ] Updated CHANGELOG.md
  > (should I do this?)

### CODE REVIEW

File: src/Iris/Tool.hs [hunk 1]
```diff
@@ -132,6 +136,32 @@ checkTool cmd Tool{..} = findExecutable (Text.unpack toolName) >>= \case
...
+        ( Exception  -- ^ @since 0.0.0.0
+        )
+
+
+{-|
+
+__Throws:__ 'ToolCheckException' if can't find a tool or if it has wrong version.
+@since 0.0.0.0
+-}
+need :: [Tool] -> IO ()
```
Comment (german1608) [inline src/Iris/Tool.hs:162]: I don't know if this is the best signature for the `need` function. Maybe we should add a `MonadIO m` constraint? Let me know, please :)

Comment (chshersh) [reply src/Iris/Tool.hs:162]: Agree, let's actually have `MonadIO m =>` constraint here. This function is supposed to be called from inside the custom monad, so having the `MonadIO` constraint allows us to do this easily without the need to call `liftIO`.

File: test/Test/Iris/Tool.hs [hunk 2]
```diff
@@ -2,40 +2,56 @@ module Test.Iris.Tool (toolSpec) where
...
+                _other                          -> False
+
+        need [Tool tool Nothing] `shouldThrow` expectedExceptionSel
+
+    it "should fail when 'need'ing tools with wrong version" $ do
+
+        let expectedExceptionSel (err :: ToolCheckException) = case err of
+                ToolCheckException (ToolWrongVersion _) -> True
+                _other                          -> False
+        need [ghc100] `shouldThrow` expectedExceptionSel
```
Comment (chshersh) [inline test/Test/Iris/Tool.hs:54]: Thanks for adding extra tests! 👏🏻

**Review by chshersh — CHANGES_REQUESTED** [review]
Hi @german1608 👋🏻 

There's one minor change in the API of `need` but the rest of the PR looks great 🙂 

Happy to merge the PR after making this change and rebasing on top of the latest `main` branch.

**Review by chshersh — APPROVED** [review]
Awesome!

File: test/Test/Iris/Tool.hs [hunk 3]
```diff
@@ -2,40 +2,56 @@ module Test.Iris.Tool (toolSpec) where
...
-                }
-
         let isToolWrongVersion :: ToolCheckResult -> Bool
             isToolWrongVersion (ToolCheckError (ToolWrongVersion _)) = True
             isToolWrongVersion _other               = False
 
-        checkTool () tool >>= (`shouldSatisfy` isToolWrongVersion)
+        checkTool ghc100 >>= (`shouldSatisfy` isToolWrongVersion)
+
+    it "should not fail when 'need'ing 'curl'" $ (need ["curl"] :: IO ())
```
Comment (chshersh) [inline test/Test/Iris/Tool.hs:39]: HLint has a warning here and thus CI is failing. I'll probably disable some of the annoying HLint warnings in the future but for now, let's just remove this extra `$` to make HLint happy
```suggestion
    it "should not fail when 'need'ing 'curl'" (need ["curl"] :: IO ())
```

**Pull request merged**